sklearn/svm/_base.py

```diff
@@ -274,6 +274,17 @@ def fit(self, X, y, sample_weight=None):
             self.intercept_ *= -1
             self.dual_coef_ = -self.dual_coef_
 
+        # check for finiteness of coefficientss
+        coef_finiteness = (
+            np.isfinite(self._dual_coef_.data).all()
+            if self._sparse
+            else np.isfinite(self._dual_coef_).all()
+        )
+        intercept_finiteness = np.isfinite(self._intercept_).all()
+        if not (coef_finiteness and intercept_finiteness):
+            # maybe check which coeff/feature is non-finite?
+            raise ValueError("Iterative parameter estimation led to non-finite values.")
```

Comment:
I think the error message can be very explicit, so users know what is wrong and what to do:

```suggestion
            raise ValueError("The dual coefficients or intercepts are not finite. "
                             "The input data may contain large values and need to be"
                             "preprocessed.")
```